core/platform/email/gae_email_services_test.py

```diff
@@ -38,8 +44,166 @@ def test_sending_email(self):
 
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             gae_email_services.send_mail(
-                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                feconf.SYSTEM_EMAIL_ADDRESS,
+                feconf.ADMIN_EMAIL_ADDRESS,
                 'subject', 'body', 'html', bcc_admin=False)
         messages = self.mail_stub.get_sent_messages(
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(1, len(messages))
+
+    def test_that_email_bcc_is_sent_if_bcc_admin_is_true(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=True)
+        messages = self.mail_stub.get_sent_messages(
+            to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertEqual(messages[0].bcc, feconf.ADMIN_EMAIL_ADDRESS)
+
+    def test_that_email_bcc_not_sent_if_bcc_admin_is_false(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=False)
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertFalse(hasattr(messages, 'bcc'))
+
+    def test_sending_email_with_reply_to_id_adds_reply_to_email(self):
+        reply_to = 'reply_to_address'
+
+        # Tests that email has reply_to_id if passed.
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                'subject', 'body', 'html', reply_to_id=reply_to)
+        messages = self.mail_stub.get_sent_messages(
+            to=feconf.ADMIN_EMAIL_ADDRESS)
+        valid_reply_to = gae_email_services.get_incoming_email_address(reply_to)
```

Comment:
Nope -- use a hardcoded string in the expectation below. Test interface, not implementation.

Reply:
Done